### ISSUE TITLE
MissingErrorInput when calling a model containing a layer with masking on new inputs

### ISSUE THREAD

**Issue by around1991** [fchollet/keras#5829]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

https://gist.github.com/around1991/cb6e8b7d17c3d1d0362e16e5a91e0dbb reliably reproduces the bug.

The issue is in line 2131 of topology.py: if mask is one of the kwargs when the layer was originally called in the inner model, then the mask kwarg is stored in layer.call, and isn't refreshed when model.call() is called.

**Comment by around1991** [comment]
Although, if like 2131 is changed to always refresh the 'mask' kwarg, then if someone (i.e. me :( ) called a layer with custom masking inside a defined Model, and then reapplies the Model to something else, it's going to break horribly.
